File: scratch/test-cleaner.ts
import { cleanPhoneticInput } from '../lib/phoneticCleaner';

const cases = [
  { input: "To me ki kaadz kor ba na", expected: "tumi ki kaaj korba na" },
  { input: "ami vat kai", expected: "ami bhaat khai" },
  { input: "tume korsi", expected: "tumi korechi" },
  { input: "dekhtesi valo gach", expected: "dekchi valo gach" }
];

console.log("Running Phonetic Cleaner Tests:\n");
let passed = true;

for (const { input, expected } of cases) {
  const got = cleanPhoneticInput(input);
  const match = got.trim().toLowerCase() === expected.trim().toLowerCase();
  console.log(`Input:    "${input}"`);
  console.log(`Expected: "${expected}"`);
  console.log(`Got:      "${got}"`);
  console.log(`Status:   ${match ? "✅ PASS" : "❌ FAIL"}\n`);
  if (!match) passed = false;
}

if (passed) {
  console.log("🎉 All phonetic cleaner tests passed successfully!");
} else {
  console.log("❌ Some phonetic cleaner tests failed.");
  process.exit(1);
}

File: scratch/test_cleaner.py
import sys
import os

sys.path.append(os.path.join(os.path.dirname(__file__), '..', 'backend'))

from phonetic_cleaner import clean_phonetic_input

cases = [
    ("To me ki kaadz kor ba na", "tumi ki kaaj korba na"),
    ("ami vat kai", "ami bhaat khai"),
    ("tume korsi", "tumi korechi"),
    ("dekhtesi valo gach", "dekchi valo gach")
]

print("Running Python Phonetic Cleaner Tests:\n")
passed = True

for raw, expected in cases:
    got = clean_phonetic_input(raw)
    match = got.strip().lower() == expected.strip().lower()
    print(f"Input:    '{raw}'")
    print(f"Expected: '{expected}'")
    print(f"Got:      '{got}'")
    print(f"Status:   {'✅ PASS' if match else '❌ FAIL'}\n")
    if not match:
        passed = False

if passed:
    print("🎉 All python phonetic cleaner tests passed successfully!")
else:
    print("❌ Some python phonetic cleaner tests failed.")
    sys.exit(1)
